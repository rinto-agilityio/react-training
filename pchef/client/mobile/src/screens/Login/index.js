import React from 'react'
import AsyncStorage from '@react-native-community/async-storage'
import { NavigationScreenProps } from 'react-navigation'

// Component
import LoginContainer from 'pchef-shared/src/containers/Login'

// Constants
import ROUTES from '@constants/routes'

type Props = {
  navigation: NavigationScreenProps,
}
// Login
const Login = ({ navigation }: Props) => {
  // save token and navigation to home screen
  const handlingLoginSuccess = async (token: string) => {
    await AsyncStorage.setItem('token', token)
    navigation.navigate(ROUTES.HOME)
  }

  return (
    <LoginContainer
      handlingLoginSuccess={handlingLoginSuccess}
      handleNavigateHome={() => navigation.navigate(ROUTES.HOME)}
      handleNavigateWelcome={() => navigation.navigate(ROUTES.WELCOME)}
    />
  )
}

export default Login
