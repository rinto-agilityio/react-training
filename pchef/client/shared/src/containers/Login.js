import { compose } from 'react-apollo'

// GraphQL
import { signInUser } from '../graphql/user'

type Props = {
  signInUser: (email: string, password: string) => void,
  children: any,
}

const LoginContainer = ({ children, signInUser }: Props) => children({ signInUser })

export default compose(
  signInUser,
)(LoginContainer)
