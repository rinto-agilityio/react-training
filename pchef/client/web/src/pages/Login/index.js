import React from 'react'

// Containers
import LoginContainer from 'pchef-shared/src/containers/Login'

type Props = {
  history: Object
}

const Login = ({ history }: Props) => {
  const handlingLoginSuccess = (token: string) => {
    localStorage.setItem('token', token)
    history.push('/')
  }

  return (
    <LoginContainer
      handlingLoginSuccess={handlingLoginSuccess}
    />
  )
}

export default Login
