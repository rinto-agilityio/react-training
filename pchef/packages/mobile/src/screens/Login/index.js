import React from 'react'
import { View, StyleSheet, Text } from 'react-native'


// define Login page styles
const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
})

// Login
const Login = () => (
  <View style={styles.container}>
    <Text>test</Text>
  </View>
)

export default Login
