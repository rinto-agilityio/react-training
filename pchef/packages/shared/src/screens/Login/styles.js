// Libs
import { StyleSheet } from 'react-native';


export const styles = StyleSheet.create({
  textBoxWrapper: {
    margin: 10,
  },
  btnWrapper: {
    padding: 10,
    width: '100%'
  }
});
