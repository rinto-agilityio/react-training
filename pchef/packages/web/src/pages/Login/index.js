import React from 'react'

// components
import { Link } from 'react-router-dom'

const Login = () => (
  <>
    <h3>Login page</h3>
    <Link to="/">Link to Home</Link>
  </>
)

export default Login
