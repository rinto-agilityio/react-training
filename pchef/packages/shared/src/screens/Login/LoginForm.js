// Libs
import React from 'react';
import { View } from 'react-native';

// Components
import TextBox from '../../components/TextBox';
import Button from '../../components/Button';
import Wrapper from '../../layout/Wrapper';

// Styles
import { styles } from './styles';

type Props = {
  customStyle?: Object
}

const LoginForm = ({ customStyle }: Props) => {
  return (
    <Wrapper direction='column' customStyle={customStyle}>
      <View style={styles.textBoxWrapper}>
        <TextBox 
          placeholder="Enter your email..." 
          placeholderTextColor="darkgray"
          customStyle={{ width: '100%'}}
          
          />
      </View>
      <View style={styles.textBoxWrapper}>
        <TextBox 
          placeholder="Enter your password..." 
          placeholderTextColor="darkgray"
          customStyle={{ width: '100%'}}
        />
      </View>
      <View style={styles.btnWrapper}>
        <Button title="Login" />
      </View>
    </Wrapper>
  );
};


export default LoginForm;
