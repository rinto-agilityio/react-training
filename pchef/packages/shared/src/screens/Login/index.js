// Libs
import React from 'react';
import { View } from 'react-native';

// Components
import LoginForm from './LoginForm';

type Props = {
  customStyle?: Object
}
const Login = ({ customStyle }: Props) => (
  <View>
    <LoginForm customStyle={customStyle} />
  </View>
);

Login.defaultProps = {
  customStyle: {},
}
export default Login;
