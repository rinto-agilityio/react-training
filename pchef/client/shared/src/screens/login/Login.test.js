// Components
import wait from 'waait'
import Login from '.'

it('Login snapshots', () => {
  // define props
  const props = {
    handlingLoginSuccess: () => {},
    signInUser: () => {},
  }

  const primaryComponent = shallow(
    <Login {...props} type="primary" customStyle={{ flex: 1 }} />,
  )
  const secondaryComponent = shallow(<Login {...props} type="secondary" />)

  expect(primaryComponent).toMatchSnapshot()
  expect(secondaryComponent).toMatchSnapshot()
})

it('Login actions', async () => {
  // define props
  const props = {
    signInUser: () => new Promise((rs, err) => {
      err()
    }),
    handlingLoginSuccess: jest.fn(),
  }

  const component = shallow(<Login {...props} />)

  // login failed
  component
    .find('LoginForm')
    .at(0)
    .props()
    .onSubmit()

  await wait(0)

  expect(component.find('Text').length).toBe(2)

  // login success
  component.setProps({
    signInUser: () => new Promise(resolve => {
      resolve({ data: { signInUser: { token: 'test' } } })
    }),
  })

  component
    .find('LoginForm')
    .at(0)
    .props()
    .onSubmit()

  await wait(0)

  expect(props.handlingLoginSuccess).toHaveBeenCalled()
})
