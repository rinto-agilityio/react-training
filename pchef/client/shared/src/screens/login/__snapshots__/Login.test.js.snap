// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login snapshots 1`] = `ShallowWrapper {}`;

exports[`Login snapshots 2`] = `ShallowWrapper {}`;
